File: Sample Code/DJISdkDemo/Demo/Battery/GetSetSelfDischargeDay/BatterySelfDischargeViewController.m
//
//  BatterySelfDischargeViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//
/**
 *  This file demonstrates how to change setting through DJIBattery.
 */
#import "BatterySelfDischargeViewController.h"
#import "DemoUtility.h"
#import <DJISDK/DJISDK.h>

@interface BatterySelfDischargeViewController ()

@end

@implementation BatterySelfDischargeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Self Discharge Days";
    self.rangeLabel.text = @"The input should be an integer. The range is [1, 10]. ";
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)onGetButtonClicked:(id)sender {
    [self updateSelfDischargeDay];
}

-(void) updateSelfDischargeDay {
    DJIBattery* battery = [DemoComponentHelper fetchBattery];
    if (battery) {
        WeakRef(target);
        [battery getSelfDischargeDayWithCompletion:^(uint8_t day, NSError * _Nullable error) {
            WeakReturn(target);
            if (error) {
                ShowResult(@"ERROR: getBatterySelfDischargeDay. %@", error.description);
            }
            else {
                NSString* getTextString = [NSString stringWithFormat:@"%u", day];
                target.getValueTextField.text = getTextString;
            }
        }];
    }
}

- (IBAction)onSetButtonClicked:(id)sender {
    DJIBattery* battery = [DemoComponentHelper fetchBattery];
    if (battery) {
        int selDischargeDay = [self.setValueTextField.text intValue];

        [battery setSelfDischargeDay:selDischargeDay withCompletion:^(NSError * _Nullable error) {
            if (error) {
                ShowResult(@"ERROR: setBatterySelfDischargeDay. %@. ", error.description);
            }
            else {
                ShowResult(@"Succeed. ");
            }
        }];
    }
}

@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/Battery/GetSetSelfDischargeDay/BatterySelfDischargeViewController.swift
//
//  BatterySelfDischargeViewController.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 12/17/15.
//  Copyright © 2015 DJI. All rights reserved.
//
import DJISDK

class BatterySelfDischargeViewController: DemoGetSetViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Self Discharge Days"
        self.rangeLabel.text = "The input should be an integer. The range is [1, 10]. "
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction override func onGetButtonClicked(_ sender: AnyObject) {
        self.updateSelfDischargeDay()
    }

    func updateSelfDischargeDay() {
        let battery: DJIBattery? = self.fetchBattery()
        if battery != nil {
            
            battery!.getSelfDischargeDay(completion: {[weak self](day: UInt8, error: Error?) -> Void in
                
                if error != nil {
                    self?.showAlertResult("ERROR: getBatterySelfDischargeDay \(error!)")
                }
                else {
                    let getTextString: String = "\(day)"
                    self?.getValueTextField.text = getTextString
                }
            })
        }
    }

    @IBAction override func onSetButtonClicked(_ sender: AnyObject) {
        let battery: DJIBattery? = self.fetchBattery()
        if (battery != nil && self.setValueTextField.text != "") {
            let selDischargeDay: UInt8 = UInt8(self.setValueTextField.text!)!
            battery!.setSelfDischargeDay(selDischargeDay, withCompletion: {[weak self](error: Error?) -> Void in
                if error != nil {
                    self?.showAlertResult("ERROR: setBatterySelfDischargeDay:\(error!)")
                }
                else {
                    self?.showAlertResult("Success. ")
                }
            })
        }
    }
}

